app/services/DateService.scala

```diff
@@ -16,28 +16,28 @@
 
 package services
 
-import auth.MtdItUser
 import com.google.inject.ImplementedBy
-import config.FrontendAppConfig
-import config.featureswitch.FeatureSwitching
+import config.{FrontendAppConfig, TimeMachine}
 import models.incomeSourceDetails.TaxYear
 
 import java.time.LocalDate
 import java.time.Month.{APRIL, JANUARY}
 import javax.inject.{Inject, Singleton}
 
 @Singleton
-class DateService @Inject()(implicit val frontendAppConfig: FrontendAppConfig) extends DateServiceInterface with FeatureSwitching {
+class DateService @Inject()(implicit val frontendAppConfig: FrontendAppConfig) extends DateServiceInterface with TimeMachine {
   val appConfig: FrontendAppConfig = frontendAppConfig
 
-  def getCurrentDate: LocalDate = {
-//    if (isEnabled(TimeMachineAddYear)) {
-//      frontendAppConfig.timeMachineAddYears.map(years =>
-//        LocalDate.now().plusYears(years)
-//      ).getOrElse(LocalDate.now())
-//    } else {
+  def now(): LocalDate = LocalDate.now()
+
+  override def getCurrentDate: LocalDate = {
+    if (getTimeMachineConfig.isTimeMachineEnabled) {
+      now()
+        .plusYears(getTimeMachineConfig.addYears)
+        .plusDays(getTimeMachineConfig.addDays)
+    } else {
       LocalDate.now()
```

Comment:
you would have to call now() in line 39, but I guess this would not fail in UT and ITs as we are using fake